### PR TITLE
issue/2504 v2 accessibility improvements

### PR DESCRIPTION
**Pull request by oliverfoster**
#2504 
* removed whitespace
* backported focus behaviour from #2453 
* maintained focus on accessibility-state update
* force + remove focusable on non-natively focusable elements

### CODE REVIEW

**Comment by moloko** [comment]
Pretty sure #2446 is an issue in v2 also, worth fixing that as part of this update?

File: src/core/js/libraries/jquery.a11y.js [hunk 1]
```diff
@@ -904,24 +904,31 @@
...
+            // preventScroll function is expecting a jquery event object.
+            return preventScroll($.event.fix(event));
+        }
 
     // PRIVATE $.a11y FUNCTIONS
         function a11y_setupScrollListeners() {
-            var scrollEventName = "wheel mousewheel";
-            $(window).on(scrollEventName, preventScroll);
-            $(document).on(scrollEventName, preventScroll);
+            window.addEventListener("wheel", nativePreventScroll, { passive: false });
```
Comment (tomgreenfield) [inline src/core/js/libraries/jquery.a11y.js]: Not supported in IE8?

File: src/core/js/accessibility.js [hunk 2]
```diff
@@ -273,22 +273,22 @@ define([
...
             var isEnabledOnTouchDevices = config._isEnabledOnTouchDevices || (config._isDisabledOnTouchDevices === false);
-            
+
             if (!Modernizr.touch || this.isActive() || !isEnabledOnTouchDevices) return;
 
             //If a touch device and not enabled, remove accessibility button and turn on accessibility
 
             this._isLoaded = true;
 
             //Force accessibility on
```
Comment (moloko) [inline src/core/js/accessibility.js]: comment now inaccurate?

**Comment by oliverfoster** [comment]
@moloko @tomgreenfield ready when you are

**Comment by moloko** [comment]
@oliverfoster hopefully I can take a look tomorrow; bit swamped today

**Comment by oliverfoster** [comment]
@moloko wants to test before approving

**Pull request merged**